src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,105 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import React, { useState, useEffect } from 'react';
+
+interface PeopleFiltersProps {
+  persons: Person[];
+  onFilterChange: (filter: 'all' | 'male' | 'female') => void;
+
+  onClick: (century: string | 'all' | null) => void;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const PeopleFilters: React.FC<PeopleFiltersProps> = ({
+  onFilterChange,
+  onClick,
+  onChange,
+}) => {
+  const [filter, setFilter] = useState<'all' | 'male' | 'female'>('all');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [searchParams, setSearchParams] = useSearchParams();
```

Comment:
The `useSearchParams` hook returns an object that behaves like a URLSearchParams instance. However, directly modifying this object (like using `set` or `delete` methods) does not trigger a re-render or update the URL. Consider using `setSearchParams` to update the search parameters.